### PR TITLE
Core: Build on ARM64 - Raspberry Pi (preferable 4)

### PR DESCRIPTION
**Pull request by damaex**
**Changes proposed:**

- used a lot of fixes from https://github.com/pablocrossa/trinitycore-portable
- updated argon2 dependency and added unoptimized reference implementation
- updated efsw dependency for aarch64 support

**Target branch(es):**

- [X] 3.3.5

**Tests performed:**

- [x] builds and runs on
    - [x] raspberry pi 3b (ubuntu 19.10 64bit)
    - [x] raspberry pi 4 (ubuntu 19.10 64bit)

```bash
apt-get install git clang cmake make gcc g++ libmariadbclient-dev libssl-dev libbz2-dev libreadline-dev libncurses-dev libboost-all-dev mariadb-server p7zip libmariadb-client-lgpl-dev-compat
```

**Known issues and TODO list:**

- [x] seg fault on startup of `worldserver`
- [x] no dynamic build yet
- [x] used std::mt19937_64 instead of external MersenneTwister.h

### CODE REVIEW

**Comment by Aokromes** [comment]
try 

apt-get install git clang cmake make gcc g++ libmariadbclient-dev libssl-dev libbz2-dev libreadline-dev libncurses-dev libboost-all-dev mariadb-server p7zip libmariadb-client-lgpl-dev-compat

instead touch mysql things.

**Comment by Ascathor** [comment]
I am not sure if I should call this ambitious or celebrate the fact someone is doing this.
Somehow I like that.

**Comment by damaex** [comment]
> try
> 
> apt-get install git clang cmake make gcc g++ libmariadbclient-dev libssl-dev libbz2-dev libreadline-dev libncurses-dev libboost-all-dev mariadb-server p7zip libmariadb-client-lgpl-dev-compat
> 
> instead touch mysql things.

Yeah, this is a better way and working.

**Comment by jackpoz** [comment]
Is this related to https://github.com/TrinityCore/TrinityCore/pull/23238 ?

File: dep/jemalloc/jemalloc_internal_defs.h.in.cmake [hunk 1]
```diff
@@ -33,7 +33,12 @@
  * Hyper-threaded CPUs may need a special instruction inside spin loops in
  * order to yield to another virtual CPU.
  */
+#ifdef RASPBERRY_PI
+#define CPU_SPINWAIT __asm__ volatile("nop")
```
Comment (jackpoz) [inline dep/jemalloc/jemalloc_internal_defs.h.in.cmake]: Is this what you get when configuring jemalloc on a Raspberry PI device ?

File: src/common/Utilities/Random.cpp [hunk 2]
```diff
@@ -17,13 +17,76 @@
 
 #include "Random.h"
 #include "Errors.h"
+
+#if defined( RASPBERRY_PI )
+#include "../mersennetwister/MersenneTwister.h"
```
Comment (Shauren) [inline src/common/Utilities/Random.cpp]: Is there a reason you add external mersenne library instead of using the one in std::?

Comment (damaex) [reply src/common/Utilities/Random.cpp]: Good idea, i will gonna do that.

**Comment by jackpoz** [comment]
Maybe efsw update could be moved to its own PR

**Comment by Shauren** [comment]
same with argon tbh

**Comment by jackpoz** [comment]
My wishlist:
- update every dependency required in its own PR
- submit custom changes for ARM for dependencies to their repository and wait for them to merge it
- include in the TC PR the bare minimum TC-related changes

As "someone" will have to maintain ARM support once added, the less custom changes we have the easier it will be.

**Comment by Shauren** [comment]
Ref points 2 and 3, if you must do have it faster, include a .diff (as file) in our repo, just like we do for g3d (see https://github.com/TrinityCore/TrinityCore/tree/3.3.5/dep/g3dlite)

**Comment by jackpoz** [comment]
Latest G3D seems to handle ARM
https://casual-effects.com/g3d/G3D10/G3D-base.lib/include/G3D-base/platform.h
```cpp
/** \def G3D_ARM 
   Defined on ARM processors */

/** \def G3D_X86
   Defined on x86 processors */
#if defined(__arm64__) || defined(__ARM_ARCH) || defined(__arm) || defined (__aarch64__)
#    define G3D_ARM
#elif defined(__x86_64__) || defined(__IA64__) || defined(_M_X64) || defined(_M_IA64)
#    define G3D_X86
#endif
```

I have an old G3D 10 branch for 335 at https://github.com/TrinityCore/TrinityCore/compare/3.3.5...jackpoz:G3D10 and "6.x" (old master) at https://github.com/TrinityCore/TrinityCore/compare/master...jackpoz:G3D10_6x . 

I could take that, update it to latest 335, update it to latest G3D10, open a PR, test it and merge it, so we won't have to apply any additional custom change for G3D.

**Comment by damaex** [comment]
Added Pull Requests for:
- [x] aarch64 definition for jemalloc
- [x] update efsw to newest version
- [x] update argon2 to latest release

**Comment by Shauren** [comment]
1. kind of sucks but acceptable
2. this would be ideal
3. is unacceptable, std:: variant is much slower
4. is unacceptable, openssl has crypto quality random generator (again, slow)
5. slow and we kind of dislike boost for being horrible on compile times

**Comment by jackpoz** [comment]
Remind me to update G3D to latest (I'll open a PR about it)

**Comment by Aokromes** [comment]
remove efsw update part from here if unneeded now, if still needed, update.

**Comment by Shauren** [comment]
There is no point to update this PR yet, there are more PRs with lib updates to deal with first

**Comment by damaex** [comment]
@jackpoz Freindly reminder for updating G3D

**Comment by jackpoz** [comment]
@damaex change of plan, we don't want to update G3D from 9 to 10, we are going to keep the current version.
Please include the required changes for ARM support in this PR.

**Comment by jackpoz** [comment]
@damaex please let us know if you are waiting for us to do anything else, otherwise I think you can rebase this branch on latest 3.3.5 and then we can merge it

**Comment by jackpoz** [comment]
@damaex still there ?

**Comment by illfated** [comment]
No GitHub activity since November 4, 2019, for @damaex -- either too busy in real life or taking a break from coding?
I hope nothing serious has happened.

**Comment by damaex** [comment]
Sry, i was on parental leave.

**Comment by illfated** [comment]
😃 Oh. That makes sense. Congrats on your family increase. Glad to see you're back and providing us with feedback. 👍

**Comment by jackpoz** [comment]
Thanks for the PR :)

**Comment by FrostbyteGR** [comment]
It's still forcing SSE2 when trying to compile, resulting in the following error types:

- unknown type name ‘__m128i’
- incompatible types when assigning to type ‘int’ from type ‘uint32x4_t’

Documentation does not contain instructions for aarch64, in case a special cmake flag is needed to make it work.

Raspberry Pi 4 (4GB) w/ SSD
cmake3 version 3.14.6
gcc (GCC) 9.1.1 20190605 (Red Hat 9.1.1-2)

LSB Version:    :core-4.1-aarch64:core-4.1-noarch
Distributor ID: CentOS
Description:    CentOS Linux release 7.7.1908 (AltArch)
Release:        7.7.1908
Codename:       AltArch

**Comment by damaex** [comment]
@FrostbyteGR try clang as your compiler, haven't tried with gcc yet
https://trinitycore.atlassian.net/wiki/spaces/tc/pages/10977288/Linux+Requirements

**Comment by damaex** [comment]
Added PR: #24216

**Comment by ibakula** [comment]
> Added PR: #24216

Sorry for going off-topic but since you ran TC on Raspberry Pi and I would probably like to do the same would you unveil the performance of running it on the Pi? For instance start time, world lag, whatever you notice slower than running on amd64 arch builds

**Comment by damaex** [comment]
> > Added PR: #24216
> 
> Sorry for going off-topic but since you ran TC on Raspberry Pi and I would probably like to do the same would you unveil the performance of running it on the Pi? For instance start time, world lag, whatever you notice slower than running on amd64 arch builds

Running now on an Raspberry Pi 4 (8GB) with Ubuntu 20.04.

It works really good! Worldserver starts (with database updates) at around 45 seconds from an sd-card with the database-server (mariadb) on the same machine. Authserver is up and running in about 5 seconds.

We tried with 10 accounts at the same time and didn't had any problems or noticed lag, but we were only in 2 instances at the same time. A full build takes some time for the first build, but updates are okay. The reload of the dynamic linked libraries works aswell. I also would recommend running the map extraction on a different, more powerfull machine with an ssd and copy the data over.

Hope this helps.

Full start log:
```
TrinityCore rev. 169ee64502f0 2020-07-08 18:46:03 +0200 (3.3.5 branch) (Unix, RelWithDebInfo, Dynamic) (worldserver-daemon)
<Ctrl-C> to stop.

 ______                       __
/\__  _\       __          __/\ \__
\/_/\ \/ _ __ /\_\    ___ /\_\ \, _\  __  __
   \ \ \/\`'__\/\ \ /' _ `\/\ \ \ \/ /\ \/\ \
    \ \ \ \ \/ \ \ \/\ \/\ \ \ \ \ \_\ \ \_\ \
     \ \_\ \_\  \ \_\ \_\ \_\ \_\ \__\\/`____ \
      \/_/\/_/   \/_/\/_/\/_/\/_/\/__/ `/___/> \
                                 C O R E  /\___/
http://TrinityCore.org                    \/__/

Using configuration file /home/ubuntu/server/etc/worldserver.conf.
Using SSL version: OpenSSL 1.1.1f  31 Mar 2020 (library: OpenSSL 1.1.1f  31 Mar 2020)
Using Boost version: 1.71.0
Updating Auth database...
>> Auth database is up-to-date! Containing 2 new and 67 archived updates.
Updating Character database...
>> Character database is up-to-date! Containing 1 new and 48 archived updates.
Updating World database...
>> Applying update "2020_07_08_00_world.sql" 'E7E831D'...
>> Applying update "2020_07_08_01_world.sql" 'A21E931'...
>> Applying update "2020_07_08_02_world.sql" '0D14B73'...
>> Applying update "2020_07_08_03_world.sql" '0D05BC6'...
>> Applying update "2020_07_08_04_world.sql" '02A8692'...
>> Applying update "2020_07_08_05_world.sql" 'D2550A5'...
>> Applying update "2020_07_08_06_world.sql" 'DC726B0'...
>> Applying update "2020_07_08_07_world.sql" '4F0C05A'...
>> Applying update "2020_07_08_08_world.sql" '402D3D8'...
>> Applying update "2020_07_08_09_world.sql" '5DA869E'...
>> Applied 10 queries. Containing 87 new and 4934 archived updates.
Realm running as realm ID 1
Using World DB: TDB 335.20061
Missing name Creature.RegenHPCannotReachTargetInRaid in config file /home/ubuntu/server/etc/worldserver.conf, add "Creature.RegenHPCannotReachTargetInRaid = 1" to this file
Will clear `logs` table of entries older than 1209600 seconds every 10 minutes.
Client cache version set to: 20061
Using DataDir /home/ubuntu/server/data/
WORLD: MMap data directory is: /home/ubuntu/server/data/mmaps
VMap support included. LineOfSight: 1, getHeight: 1, indoorCheck: 1
VMap data directory is: /home/ubuntu/server/data/vmaps
Loading Trinity strings...
>> Loaded 967 trinity strings in 6 ms
Initialize data stores...
>> Initialized 113 data stores in 4711 ms
Using enUS DBC Locale as default. All available DBC locales: deDE enUS
>> Loading Cinematic Camera files
>> Loaded 0 cinematic waypoint sets in 0 ms
Loading IP Location Database...
Initializing PlayerDump tables...
>> Initialized tables for PlayerDump in 48 ms.
Loading SpellInfo store...
>> Loaded SpellInfo store in 154 ms
Loading SpellInfo corrections...
>> Loaded SpellInfo corrections in 19 ms
Loading SkillLineAbilityMultiMap Data...
>> Loaded 10219 SkillLineAbility MultiMap Data in 3 ms
Loading SpellInfo custom attributes...
>> Loaded 264 spell custom attributes from DB in 6 ms
>> Loaded SpellInfo custom attributes in 76 ms
Loading SpellInfo diminishing infos...
>> Loaded SpellInfo diminishing infos in 13 ms
Loading SpellInfo immunity infos...
>> Loaded SpellInfo immunity infos in 28 ms
Loading Player Totem models...
>> Loaded 20 player totem model records in 4 ms
Loading GameObject models...
Loading Script Names...
>> Loaded 2995 ScriptNames in 4595 ms
Loading Instance Template...
>> Loaded 84 instance templates in 1 ms
Loading instances...
>> Loaded instances in 6 ms
Loading character cache store...
Loaded character infos for 5 characters in 0 ms
Loading Broadcast texts...
>> Loaded 73039 broadcast texts in 429 ms
>> Loaded 73039 broadcast text locales in 2693 ms
Loading Localization strings...
>> Loaded 27246 creature locale strings in 524 ms
>> Loaded 19673 gameobject_template_locale strings in 359 ms
>> Loaded 38113 Item locale strings in 824 ms
>> Loaded 2472 Item set name locale strings in 40 ms
>> Loaded 9456 Quest locale strings in 990 ms
>> Loaded 1748 Quest Offer Reward locale strings in 18 ms
>> Loaded 860 Quest Request Items locale strings in 8 ms
>> Loaded 25 NpcText locale strings in 6 ms
>> Loaded 1946 PageText locale strings in 126 ms
>> Loaded 1 gossip_menu_option locale strings in 5 ms
>> Loaded 453 points_of_interest locale strings in 14 ms
>> Loaded 14 quest greeting locale strings in 5 ms
>> Localization strings loaded in 2920 ms
Loading Account Roles and Permissions...
>> Loaded 670 permission definitions, 648 linked permissions and 4 default permissions in 7 ms
Loading Page Texts...
>> Loaded 1950 page texts in 30 ms
Loading Game Object Templates...
>> Loaded 21572 game object templates in 268 ms
Loading Game Object template addons...
>> Loaded 21500 game object template addons in 52 ms
Loading Transport templates...
>> Loaded 30 transport templates in 30 ms
Loading Transport animations and rotations...
Loading Spell Rank Data...
>> Loaded 3465 spell rank records in 17 ms
Loading Spell Required Data...
>> Loaded 41 spell required records in 5 ms
Loading Spell Group types...
>> Loaded 549 spell group definitions in 3 ms
Loading Spell Learn Skills...
>> Loaded 103 Spell Learn Skills from DBC in 6 ms
Loading SpellInfo SpellSpecific and AuraState...
>> Loaded SpellInfo SpellSpecific and AuraState in 18 ms
Loading Spell Learn Spells...
>> Loaded 8 spell learn spells + 574 found in DBC in 10 ms
Loading Spell Proc conditions and data...
>> Loaded 845 spell proc conditions and data in 11 ms
Generating spell proc data from SpellMap...
>> Generated spell proc data for 1386 spells in 11 ms
Loading Spell Bonus Data...
>> Loaded 216 extra spell bonus data in 5 ms
Loading Aggro Spells Definitions...
>> Loaded 106 SpellThreatEntries in 3 ms
Loading Spell Group Stack Rules...
>> Loaded 65 spell group stack rules in 3 ms
>> Parsing SPELL_GROUP_STACK_RULE_EXCLUSIVE_SAME_EFFECT stack rules...
>> Parsed 14 SPELL_GROUP_STACK_RULE_EXCLUSIVE_SAME_EFFECT stack rules in 1 ms
Loading NPC Texts...
>> Loaded 8292 npc texts in 262 ms
Loading Enchant Spells Proc datas...
>> Loaded 42 enchant proc data definitions in 5 ms
Loading Item Random Enchantments Table...
>> Loaded 28607 Item Enchantment definitions in 57 ms
Loading Disables
>> Loaded 859 disables in 7 ms
Loading Items...
>> Loaded 38607 item templates in 1723 ms
Loading Item set names...
>> Loaded 2481 item set names in 14 ms
Loading Creature Model Based Info Data...
>> Loaded 24143 creature model based info in 74 ms
Loading Creature templates...
>> Loaded 1735 creature template resistances in 8 ms
>> Loaded 9641 creature template spells in 18 ms
>> Loaded 29923 creature definitions in 922 ms
Loading Equipment templates...
>> Loaded 10863 equipment templates in 41 ms
Loading Creature template addons...
>> Loaded 11761 creature template addons in 48 ms
Loading Reputation Reward Rates...
>> Loaded 16 reputation_reward_rate in 5 ms
Loading Creature Reputation OnKill Data...
>> Loaded 2028 creature award reputation definitions in 17 ms
Loading Reputation Spillover Data...
>> Loaded 26 reputation_spillover_template in 5 ms
Loading Points Of Interest Data...
>> Loaded 453 Points of Interest definitions in 9 ms
Loading Creature Base Stats...
>> Loaded 400 creature base stats in 18 ms
Loading Spawn Group Templates...
>> Loaded 62 spawn group templates in 5 ms
Loading instance spawn groups...
>> Loaded 55 instance spawn groups in 3 ms
Loading Creature Data...
>> Loaded 151078 creatures in 2793 ms
Loading Temporary Summon Data...
>> Loaded 1022 temp summons in 7 ms
Loading pet levelup spells...
>> Loaded 993 pet levelup and default spells for 39 families in 9 ms
Loading pet default spells additional to levelup spells...
>> Loaded addition spells for 111 pet spell data entries in 6 ms
Loading summonable creature templates...
>> Loaded 460 summonable creature templates in 7 ms
Loading Creature Addon Data...
>> Loaded 36937 creature addons in 130 ms
Loading Creature Movement Overrides...
>> Loaded 10 movement overrides in 4 ms
Loading Gameobject Data...
>> Loaded 92261 gameobjects in 2043 ms
Loading Spawn Group Data...
>> Loaded 96825 spawn group members in 159 ms
Loading GameObject Addon Data...
>> Loaded 59667 gameobject addons in 232 ms
Loading GameObject faction and flags overrides...
>> Loaded 20 gameobject faction and flags overrides in 6 ms
Loading GameObject Quest Items...
>> Loaded 857 gameobject quest items in 5 ms
Loading Creature Quest Items...
>> Loaded 4391 creature quest items in 19 ms
Loading Creature Linked Respawn...
>> Loaded 4786 linked respawns in 21 ms
Loading Weather Data...
>> Loaded 35 weather definitions in 1 ms
Loading Quests...
>> Loaded 9464 quests definitions in 582 ms
Checking Quest Disables
>> Checked 687 quest disables in 0 ms
Loading Quest POI
>> Loaded 8953 quest POI definitions in 203 ms
Loading Quests Starters and Enders...
Loading GO Start Quest Data...
>> Loaded 446 quest relations from gameobject_queststarter in 4 ms
Loading GO End Quest Data...
>> Loaded 457 quest relations from gameobject_questender in 3 ms
Loading Creature Start Quest Data...
>> Loaded 7425 quest relations from creature_queststarter in 23 ms
Loading Creature End Quest Data...
>> Loaded 7860 quest relations from creature_questender in 27 ms
Loading Quests Greetings...
>> Loaded 166 quest_greeting in 6 ms
Loading Objects Pooling Data...
>> Loaded 3447 objects pools in 16 ms
Loading Creatures Pooling Data...
>> Loaded 1293 creatures in pools in 17 ms
Loading Gameobject Pooling Data...
>> Loaded 25818 gameobject in pools in 290 ms
Loading Mother Pooling Data...
>> Loaded 2568 pools in mother pools in 28 ms
Starting objects pooling system...
Loading Quest Pooling Data...
>> Loaded 35 daily, 7 weekly and 0 monthly quest pools in 9 ms
Loading Game Event Data...
>> Loaded 156 holiday dates in 2 ms
>> Loaded 83 game events in 7 ms.
Loading Game Event Saves Data...
>> Loaded 0 game event saves in game events. DB table `game_event_save` is empty.
Loading Game Event Prerequisite Data...
>> Loaded 0 game event prerequisites in game events. DB table `game_event_prerequisite` is empty.
Loading Game Event Creature Data...
>> Loaded 9837 creatures in game events in 14 ms.
Loading Game Event GO Data...
>> Loaded 31249 gameobjects in game events in 52 ms.
Loading Game Event Model/Equipment Change Data...
>> Loaded 9 model/equipment changes in game events in 5 ms.
Loading Game Event Quest Data...
>> Loaded 224 quests additions in game events in 3 ms.
Loading Game Event GO Quest Data...
>> Loaded 78 quests additions in game events in 2 ms.
Loading Game Event Quest Condition Data...
>> Loaded 0 quest event conditions in game events. DB table `game_event_quest_condition` is empty.
Loading Game Event Condition Data...
>> Loaded 0 conditions in game events. DB table `game_event_condition` is empty.
Loading Game Event Condition Save Data...
>> Loaded 0 condition saves in game events. DB table `game_event_condition_save` is empty.
Loading Game Event NPCflag Data...
>> Loaded 5 npcflags in game events in 3 ms.
Loading Game Event Seasonal Quest Relations...
>> Loaded 690 quests additions in game events in 22 ms.
Loading Game Event Vendor Additions Data...
>> Loaded 14 vendor additions in game events in 4 ms.
Loading Game Event Battleground Holiday Data...
>> Loaded 6 battleground holidays in game events in 3 ms.
Loading Game Event Pool Data...
>> Loaded 9 pools for game events in 1 ms.
Loading UNIT_NPC_FLAG_SPELLCLICK Data...
>> Loaded 325 spellclick definitions in 9 ms
Loading Vehicle Template Accessories...
>> Loaded 197 Vehicle Template Accessories in 8 ms
Loading Vehicle Accessories...
>> Loaded 0 Vehicle Accessories in 2 ms
Loading Vehicle Seat Addon Data...
>> Loaded 12 Vehicle Seat Addon entries in 2 ms
Loading SpellArea Data...
>> Loaded 766 spell area requirements in 7 ms
Loading AreaTrigger definitions...
>> Loaded 273 area trigger teleport definitions in 5 ms
Loading Access Requirements...
>> Loaded 121 access requirement definitions in 4 ms
Loading Quest Area Triggers...
>> Loaded 60 quest trigger points in 3 ms
Loading Tavern Area Triggers...
>> Loaded 114 tavern triggers in 2 ms
Loading AreaTrigger script names...
>> Loaded 194 areatrigger scripts in 1 ms
Loading LFG entrance positions...
>> Loaded 20 lfg entrance positions in 3 ms
Loading Dungeon boss data...
>> Loaded 612 instance encounters in 5 ms
Loading LFG rewards...
>> Loaded 15 lfg dungeon rewards in 3 ms
Loading Graveyard-zone links...
>> Loaded 700 graveyard-zone links in 4 ms
Loading spell pet auras...
>> Loaded 47 spell pet auras in 3 ms
Loading Spell target coordinates...
>> Loaded 585 spell teleport coordinates in 10 ms
Loading enchant custom attributes...
>> Loaded 32 custom enchant attributes in 1 ms
Loading linked spells...
>> Loaded 456 linked spells in 6 ms
Loading Player Create Data...
>> Loaded 62 player create definitions in 4 ms
Loading Player Create Items Data...
>> Loaded 1 custom player create items in 4 ms
Loading Player Create Skill Data...
>> Loaded 1094 player create skills in 5 ms
Loading Player Create Spell Data...
>> Loaded 2680 custom player create spells in 6 ms
Loading Player Create Cast Spell Data...
>> Loaded 19 player create cast spells in 3 ms
Loading Player Create Action Data...
>> Loaded 325 player create actions in 5 ms
Loading Player Create Level HP/Mana Data...
>> Loaded 800 level health/mana definitions in 7 ms
Loading Player Create Level Stats Data...
>> Loaded 4960 level stats definitions in 18 ms
Loading Player Create XP Data...
>> Loaded 79 xp for level definitions in 4 ms
Loading Exploration BaseXP Data...
>> Loaded 80 BaseXP definitions in 3 ms
Loading Pet Name Parts...
>> Loaded 313 pet name parts in 4 ms
Loading the max pet number...
>> Loaded the max pet number: 9 in 0 ms
Loading pet level stats...
>> Loaded 2560 level pet stats definitions in 18 ms
Loading Player level dependent mail rewards...
>> Loaded 25 level dependent mail rewards in 5 ms
Loading creature loot templates...
>> Loaded 318198 creature loot templates in 1284 ms
Loading fishing loot templates...
>> Loaded 281 fishing loot templates in 6 ms
Loading gameobject loot templates...
>> Loaded 16756 gameobject loot templates in 77 ms
Loading item loot templates...
>> Loaded 4998 item loot templates in 119 ms
Loading mail loot templates...
>> Loaded 117 mail loot templates in 8 ms
Loading milling loot templates...
>> Loaded 45 milling loot templates in 10 ms
Loading pickpocketing loot templates...
>> Loaded 11048 pickpocketing loot templates in 246 ms
Loading skinning loot templates...
>> Loaded 1898 skinning loot templates in 56 ms
Loading disenchanting loot templates...
>> Loaded 124 disenchanting loot templates in 15 ms
Loading prospecting loot templates...
>> Loaded 36 prospecting loot templates in 12 ms
Loading spell loot templates...
>> Loaded 166 spell loot templates in 18 ms
Loading reference loot templates...
>> Loaded refence loot templates in 328 ms
Loading Skill Discovery Table...
>> Loaded 347 skill discovery definitions in 16 ms.
Loading Skill Extra Item Table...
>> Loaded 225 spell specialization definitions in 5 ms
Loading Skill Perfection Data Table...
>> Loaded 72 spell perfection definitions in 3 ms.
Loading Skill Fishing base level requirements...
>> Loaded 112 areas for fishing base skill level in 4 ms
Loading Achievements...
>> Loaded 43 achievement references in 1 ms.
Loading Achievement Criteria Lists...
>> Loaded 7655 achievement criteria in 6 ms.
Loading Achievement Criteria Data...
>> Loaded 2693 additional achievement criteria data in 8 ms
Loading Achievement Rewards...
>> Loaded 112 achievement rewards in 5 ms.
Loading Achievement Reward Locales...
>> Loaded 19 achievement reward locale strings in 3 ms.
Loading Completed Achievements...
>> Loaded 43 realm first completed achievements in 1 ms.
Loading Item Auctions...
>> Loaded 0 auction items. DB table `auctionhouse` or `item_instance` is empty!
Loading Auctions...
>> Loaded 0 auctions. DB table `auctionhouse` is empty.
Loading Guilds...
Loading guilds definitions...
>> Loaded 0 guild definitions. DB table `guild` is empty.
Loading ArenaTeams...
>> Loaded 0 arena teams. DB table `arena_team` is empty!
Loading Groups...
>> Loaded 0 group definitions. DB table `groups` is empty!
Loading ReservedNames...
>> Loaded 0 reserved player names. DB table `reserved_name` is empty!
Loading GameObjects for quests...
>> Loaded 1798 GameObjects for quests in 5 ms
Loading BattleMasters...
>> Loaded 157 battlemaster entries in 7 ms
Loading GameTeleports...
>> Loaded 1432 GameTeleports in 12 ms
Loading Trainers...
>> Loaded 126 Trainers in 44 ms
Loading Creature default trainers...
>> Loaded 806 default trainers in 4 ms
Loading Gossip menu...
>> Loaded 6076 gossip_menu IDs in 21 ms
Loading Gossip menu options...
>> Loaded 4328 gossip_menu_option entries in 68 ms
Loading Vendors...
>> Loaded 37577 Vendors in 113 ms
Loading Waypoints...
>> Loaded 134328 waypoints in 934 ms
Loading SmartAI Waypoints...
>> Loaded 768 SmartAI waypoint paths (total 10019 waypoints) in 33 ms
Loading Creature Formations...
>> Loaded 1998 creatures in formations in 7 ms
Loading World States...
>> Loaded 95 world states in 1 ms
Loading Conditions...
>> Loaded 14418 conditions in 88 ms
Loading faction change achievement pairs...
>> Loaded 125 faction change achievement pairs in 5 ms
Loading faction change spell pairs...
>> Loaded 111 faction change spell pairs in 2 ms
Loading faction change quest pairs...
>> Loaded 251 faction change quest pairs in 3 ms
Loading faction change item pairs...
>> Loaded 1355 faction change item pairs in 9 ms
Loading faction change reputation pairs...
>> Loaded 16 faction change reputation pairs in 5 ms
Loading faction change title pairs...
>> Loaded 22 faction change title pairs in 3 ms
Loading GM tickets...
>> Loaded 0 GM tickets. DB table `gm_ticket` is empty!
Loading GM surveys...
>> Loaded GM Survey count from database in 1 ms
Loading client addons...
>> Loaded 23 known addons in 1 ms
Returning old mails...
>> No expired mails found.
Loading Autobroadcasts...
>> Loaded 0 autobroadcasts definitions. DB table `autobroadcast` is empty for this realm!
Loading spell_scripts...
>> Loaded 157 script definitions in 2 ms
Loading event_scripts...
>> Loaded 527 script definitions in 4 ms
Loading waypoint_scripts...
>> Loaded 725 script definitions in 4 ms
Loading spell script names...
>> Loaded 2217 spell script names in 8 ms
Loading Creature Texts...
>> Loaded 16486 creature texts for 3720 creatures in 102 ms
Loading Creature Text Locales...
>> Loaded 3 creature localized texts in 5 ms
Initializing Scripts...
Loading Script Waypoints for 55 creature(s)...
>> Loaded 1422 Script Waypoint nodes in 20 ms
>> Loaded spline chain data for 124 chains, consisting of 325 splines with 1684 waypoints in 20 ms
Loading C++ scripts
Correcting your CMAKE_INSTALL_PREFIX in "/home/ubuntu/projects/TrinityCore/build"...
>> CMAKE_INSTALL_PREFIX is equal to the current path of execution.
Loading script module "libscripts_kalimdor.so"...
>> Loaded script module "libscripts_kalimdor.so" ("Kalimdor" - 169ee64).
Loading script module "libscripts_custom.so"...
>> Loaded script module "libscripts_custom.so" ("Custom" - 169ee64).
Loading script module "libscripts_pet.so"...
>> Loaded script module "libscripts_pet.so" ("Pet" - 169ee64).
Loading script module "libscripts_northrend.so"...
>> Loaded script module "libscripts_northrend.so" ("Northrend" - 169ee64).
Loading script module "libscripts_outdoorpvp.so"...
>> Loaded script module "libscripts_outdoorpvp.so" ("OutdoorPvP" - 169ee64).
Loading script module "libscripts_easternkingdoms.so"...
>> Loaded script module "libscripts_easternkingdoms.so" ("EasternKingdoms" - 169ee64).
Loading script module "libscripts_spells.so"...
>> Loaded script module "libscripts_spells.so" ("Spells" - 169ee64).
Loading script module "libscripts_world.so"...
>> Loaded script module "libscripts_world.so" ("World" - 169ee64).
Loading script module "libscripts_outland.so"...
>> Loaded script module "libscripts_outland.so" ("Outland" - 169ee64).
Loading script module "libscripts_events.so"...
>> Loaded script module "libscripts_events.so" ("Events" - 169ee64).
Loading script module "libscripts_commands.so"...
>> Loaded script module "libscripts_commands.so" ("Commands" - 169ee64).
Loading script module "libscripts_battlefield.so"...
>> Loaded script module "libscripts_battlefield.so" ("Battlefield" - 169ee64).
>> Loaded 12 script modules.
>> Library reloader is listening on "/home/ubuntu/server/bin/scripts".
>> Loaded 3039 C++ scripts in 1333 ms
Validating spell scripts...
>> Validated 2810 scripts in 9 ms
Loading SmartAI scripts...
>> Loaded SmartAIMgr Helpers in 16 ms
>> Loaded 12874 SmartAI scripts in 430 ms
Loading Calendar data...
>> Loaded 0 calendar events in 0 ms
>> Loaded 0 calendar invites in 0 ms
Loading Petitions...
>> Loaded 0 petitions.
Loading Signatures...
>> Loaded 0 Petition signs!
Loading Item loot...
>> Loaded 0 stored item loots
>> Loaded 0 stored item money
Initialize query data...
Initialize game time and timers
Mail timer set to: 960, mail return is called every 1440 minutes
Starting Map System
Starting Game Event system...
Initialize AuctionHouseBot...
Initializing chat channels...
>> Loaded 0 custom chat channels. DB table `channels` is empty.
Initializing Opcodes...
Starting Arena Season...
Starting Battleground System
>> Loaded 11 battlegrounds in 0 ms
Starting Outdoor PvP System
>> Loaded 6 outdoor PvP definitions in 1 ms
Starting Battlefield System
>> Loaded 1 battlefields in 823 ms
Loading Transports...
>> Spawned 20 continent transports in 2242 ms
Loading Warden Checks...
Loading Warden Action Overrides...
Deleting expired bans...
Initializing quest reset times...
Calculate random battleground reset time...
Calculate deletion of old calendar events time...
Calculate guild limitation(s) reset time...
World initialized in 0 minutes 38 seconds
TrinityCore rev. 169ee64502f0 2020-07-08 18:46:03 +0200 (3.3.5 branch) (Unix, RelWithDebInfo, Dynamic) (worldserver-daemon) ready...
```

**Comment by Jiyuuneko** [comment]
Is it possible to run the master version on the Raspberry Pi？ : )

**Comment by illfated** [comment]
@Jiyuuneko : in theory, it should not be a problem. Practically speaking, I don't think the master branch has gotten this far in cherry-picking from the 3.3.5 branch (I could be wrong). As long as Raspberry Pi support is added (maybe you know how to do it yourself?), it should be just as easy as with the 3.3.5 branch. Please check the commit log for master to find out more.

**Comment by Shauren** [comment]
No, not yet. Still in 2017

**Pull request merged**